### ISSUE TITLE
Implement max-length config for TextLine input type

### ISSUE THREAD

**Issue by alansemenov** [enonic/xp#5335]
```
    <input name="requiredTextLine" type="TextLine">
      <label>Required Text Line</label>
      <indexed>true</indexed>
      <custom-text>Custom text</custom-text>
      <occurrences minimum="1" maximum="0"/>
      <help-text>Some help text</help-text>
      <config>
        <max-length>10</max-length>
      </config>
    </input>
```
Max number of characters will be shown under the textbox, right-aligned, and will change (decrease/increase) as user is typing inside the field (ala-Twitter). 

![image](https://user-images.githubusercontent.com/11612490/28120915-c233062c-6719-11e7-8b7b-0094a640a3db.png)

![image](https://user-images.githubusercontent.com/11612490/28121335-0436c1e8-671b-11e7-80ff-234f0035dccc.png)

It should still be possible to continue typing in the field even once the limit is reached (but the counter should not go below 0). If user is trying to save a longer text than allowed, the validation should kick in:

![image](https://user-images.githubusercontent.com/11612490/28120788-5c8bbc9c-6719-11e7-9097-d4897299bab7.png)

**Comment by aro** [comment]
Shouldn't we also have some kind of backend validation? 

I think that if a content is created via API and the value set to something longer than the max, validation should fail and it should not be possible to publish it using JS API neither.

**Comment by vbradnitski** [comment]
I have it stashed already, just wanted to discuss it with @alansemenov

**Comment by vbradnitski** [comment]
implemented in #5399
